File: task_equivalents/llm_translations/gemini/base/r/statistical_modelling_analysis/02_logistic_regression_odds_ratios.R
Weekly <- read.csv('data/csv/Weekly.csv')

x01 <- Weekly[, 3:8]
y01 <- ifelse(Weekly$Direction == 'Up', 1, 0)

df <- data.frame(y01 = y01, x01)
glm0_fit <- glm(y01 ~ ., data = df, family = binomial)
print(summary(glm0_fit))

prob <- predict(glm0_fit, type = "response")
pred <- ifelse(prob > 0.5, 1, 0)
print(table(y01, pred))
print(mean(pred == y01))

File: task_equivalents/llm_translations/gemini/base/python/statistical_modelling_analysis/02_logistic_regression_odds_ratios.py
import pandas as pd
import numpy as np
import statsmodels.api as sm

Weekly = pd.read_csv('data/csv/Weekly.csv', index_col=0)

x01 = Weekly.iloc[:, 2:8]
y01 = np.where(Weekly['Direction'] == 'Up', 1, 0)

X = sm.add_constant(x01)
glm0_fit = sm.Logit(y01, X).fit(disp=False)
print(glm0_fit.summary())

prob = glm0_fit.predict(X)
pred = np.where(prob > 0.5, 1, 0)

print(pd.crosstab(y01, pred, rownames=['Actual'], colnames=['Predicted']))
print(np.mean(pred == y01))